drone_flightplan/waypoints.py

```diff
@@ -2,345 +2,147 @@
 import argparse
 import pyproj
 import geojson
-from shapely.geometry import Polygon
-from drone_flightplan import calculate_parameters as cp
+from shapely.geometry import Point, shape
+from shapely.affinity import rotate
+from shapely.ops import transform
+from drone_flightplan.calculate_parameters import calculate_parameters as cp
 
-
-# Instantiate logger
 log = logging.getLogger(__name__)
 
 
-def generate_waypoints_within_polygon(
-    aoi: Polygon,
-    distance_between_lines_x: float,
-    distance_between_lines_y: float,
-    generate_each_points: bool,
-    generate_3d: bool,
-):
-    minx, miny, maxx, maxy = aoi.bounds
-
-    centroid = aoi.centroid
-
-    waypoints = [
-        {
-            "coordinates": (centroid.x, centroid.y),
-            "angle": "0",
-            "take_photo": False,
-            "gimbal_angle": "-90",
-        }
-    ]
-
-    # Generate waypoints within the polygon
-    y = miny
-    row_count = 0
-    angle = -90
+def generate_grid_in_aoi(aoi_polygon, x_spacing, y_spacing):
+    minx, miny, maxx, maxy = aoi_polygon.bounds
+    xpoints = int((maxx - minx) / x_spacing) + 1
+    ypoints = int((maxy - miny) / y_spacing) + 1
+
+    points = []
+    for yi in range(ypoints):
+        for xi in range(xpoints):
+            x = minx + xi * x_spacing
+            y = miny + yi * y_spacing
+            point = Point(x, y)
+            if aoi_polygon.contains(point):
+                points.append(point)
+            # Add only unique points that are near the edges of the polygon
+            offset_point = Point(x + x_spacing, y)
+            if (
+                aoi_polygon.contains(offset_point)
+                or aoi_polygon.distance(offset_point) <= x_spacing / 3
+            ):
+                points.append(offset_point)
+
+    return points
+
+
+def create_path(points, forward_spacing, generate_3d=False):
```

Comment:
Docstrings are also far from essential, but a simple description of what a function does can really help in the future!

```python
def create_path(...):
    """Given a grid of waypoints, creates a path between to output waylines"""
    ...
```

Reply:
Thanks! I've now included docstrings in the functions.